### ISSUE TITLE
Extra newline before input

### ISSUE THREAD

**Issue by arcanemutterings** [spyder-ide/spyder#5242]
## Description
This code https://gist.github.com/Nixerrr/6ab82500b7995a1b1dc70998fdf5a893 produces an unexpected output, because there's an extra newline before the input line.  
![image](https://user-images.githubusercontent.com/30010072/30470630-c10a017e-99f5-11e7-818f-2274b5fa6e07.png)
Meanwhile, this is what the same code produces in the Cloud9 IDE:
![image](https://user-images.githubusercontent.com/30010072/30470729-3420014a-99f6-11e7-91e3-329ef002afa2.png)

**What steps will reproduce the problem?**

1.  Run that code in Spyder.
2.  Run it in a different IDE
3. The output is different

**What is the expected output? What do you see instead?**
I expected:
foo!
bar? h
baz!

Instead I got:
foo!

bar? h
baz!

**Please provide any additional information below**

## Version and main components

* Spyder Version: 3.1.4 
* Python Version: 3.6.1
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.5.0 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
nbconvert >=4.0  :  5.1.1 (OK)
pandas >=0.13.1  :  0.20.1 (OK)
numpy >=1.7      :  1.12.1 (OK)
sphinx >=0.6.6   :  1.5.6 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
matplotlib >=1.0 :  2.0.2 (OK)
sympy >=0.7.3    :  1.0 (OK)
pylint >=0.25    :  1.6.4 (OK)

```

**Comment by ccordoba12** [comment]
Thanks for reporting. Although annoying, this is a simple bug and we don't have time to fix it right now. Sorry.

**Comment by mjpieters** [comment]
The linked gist is no longer available. Can the issue be updated with an inline case? From [this Stack Overflow question](https://stackoverflow.com/q/53133795/100297), I surmise it is something as simple as:

```python
print('foo!')
input('bar? ')
print('baz!')
```

**Comment by ccordoba12** [comment]
It seems an `input` statement, combined with other `print` ones generates this error.

**Comment by ccordoba12** [comment]
Closing as a duplicate of issue #16504, which was fixed on Spyder 5.3.1.
